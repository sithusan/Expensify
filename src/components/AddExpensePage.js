import React from 'react';
const AddExpensePage = () => (
    <div>
        This is for my Add Expense Component
    </div>
)
export default AddExpensePage