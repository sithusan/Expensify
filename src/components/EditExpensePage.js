import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './expenseForm'
import { RemoveExpense,editExpense } from '../actions/expenses.js'

const EditExpensePage = (props) => (
    <div>
        <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
            props.dispatch(editExpense(props.match.params.id,expense))
            props.history.push('/')
        }}
        />
        <button onClick={() => {
            props.dispatch(RemoveExpense({id:props.expense.id}))
            props.history.push('/')

        }}>Remove</button>
    </div>
)

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}
export default connect(mapStateToProps)(EditExpensePage);