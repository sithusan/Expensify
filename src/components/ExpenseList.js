import React from 'react';
import ExpenseListItem from './ExpenseListItem'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>There is no Expense</p>
            ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem
                            key={expense.id}
                            {...expense}
                        />
                    })
                )
        }
    </div>
)
const mapstateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter)
    }
}
export default connect(mapstateToProps)(ExpenseList);