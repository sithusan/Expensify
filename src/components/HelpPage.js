import React from 'react';
const HelpPage = () => (
    <div>
        This is For Help Expense Component
    </div>
)
export default HelpPage