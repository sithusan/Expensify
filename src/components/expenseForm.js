import React from 'react'
import moment from 'moment'
import 'react-dates/lib/css/_datepicker.css'

export default class ExpenseFrom extends React.Component {
    constructor(props) {
        super(props);
        { console.log(props.expense) }
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        }
    }

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }))
    }
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }))
    }
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }))
        }
    }
    onDateChange = (e) => {
        const createdAt = moment(e.target.value);
        this.setState(() => ({ createdAt }))
    }
    onFocusChange = () => {
        this.setState((prevState) => ({ calendarFocused: !prevState.calendarFocused }))
    }
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: 'Please provide description and amount' }))
        } else {
            this.setState(() => ({ error: '' }))
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    }
    render() {
        return (
            <div>
                {
                    this.state.error &&
                    <p>{this.state.error}</p>
                }
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <input
                        type="date"
                        onChange={this.onDateChange}
                    />
                    <textarea onChange={this.onNoteChange}
                        placeholder="A note for your expense"
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}