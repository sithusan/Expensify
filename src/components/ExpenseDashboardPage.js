import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './expenseListFilters'
const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList />
    </div>
)
export default ExpenseDashboardPage 