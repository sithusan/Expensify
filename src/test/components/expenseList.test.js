import { ExpenseList } from '../../components/ExpenseList'
import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'

test('Should render expenselist with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})


test('should render expense list with empty message',() => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
})