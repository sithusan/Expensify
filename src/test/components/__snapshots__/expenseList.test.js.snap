// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenselist with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="GUM"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={19500}
    createdAt={-1000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={1000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div>
  <p>
    There is no Expense
  </p>
</div>
`;
